src/server/pkg/testutil/auth.go

```diff
@@ -17,315 +14,60 @@ import (
-	}
-	return false
-}
-
-// getPachClientInternal is a helper function called by getPachClient. It
-// returns (or creates) a pach client authenticated as 'subject', but doesn't
-// do any any checks to confirm that auth is activated and the cluster is
-// configured correctly (those are done by getPachClient). If subject has no
-// prefix, they are assumed to be a GitHub user.
-//
-// Caller must hold tokenMapMut. Currently only called by getPachClient()
-func getPachClientInternal(tb testing.TB, subject string) *client.APIClient {
-	// copy seed, so caller can safely modify result
-	resultClient := seedClient.WithCtx(context.Background())
-	if subject == "" {
-		return resultClient // anonymous client
-	}
-	if subject == AdminUser {
-		resultClient.SetAuthToken(RootToken)
-		return resultClient
-	}
-
-	if token, ok := tokenMap[subject]; ok {
-		resultClient.SetAuthToken(token)
-		return resultClient
-	}
+// ActivateAuth activates the auth service in the test cluster, if it isn't already enabled
+func ActivateAuth(tb testing.TB) {
+	tb.Helper()
+	client := GetPachClient(tb)
 
-	if !strings.Contains(subject, ":") {
-		subject = gh(subject)
-	}
-	colonIdx := strings.Index(subject, ":")
-	prefix := subject[:colonIdx+1]
-	switch prefix {
-	case auth.RobotPrefix:
-		adminClient := getPachClientInternal(tb, AdminUser)
-		resp, err := adminClient.GetAuthToken(adminClient.Ctx(), &auth.GetAuthTokenRequest{
-			Subject: subject,
-		})
-		require.NoError(tb, err)
-		tokenMap[subject] = resp.Token
-	case auth.GitHubPrefix:
-		resp, err := seedClient.Authenticate(context.Background(),
-			&auth.AuthenticateRequest{
-				// When Pachyderm is deployed locally, GitHubToken automatically
-				// authenicates the caller as a GitHub user whose name is equal to the
-				// provided token
-				GitHubToken: strings.TrimPrefix(subject, auth.GitHubPrefix),
-			})
-		require.NoError(tb, err)
-		tokenMap[subject] = resp.PachToken
-	default:
-		tb.Fatalf("can't give you a client of type %s", prefix)
-	}
-	resultClient.SetAuthToken(tokenMap[subject])
-	return resultClient
-}
+	require.NoError(tb, ActivateEnterprise(tb, client))
 
-// activateAuth activates the auth service in the test cluster
-//
-// Caller must hold tokenMapMut. Currently only called by getPachClient()
-func activateAuth(tb testing.TB) {
-	resp, err := seedClient.AuthAPIClient.Activate(context.Background(),
+	_, err := client.Activate(client.Ctx(),
 		&auth.ActivateRequest{RootToken: RootToken},
 	)
 	if err != nil && !strings.HasSuffix(err.Error(), "already activated") {
 		tb.Fatalf("could not activate auth service: %v", err.Error())
 	}
-	tokenMap[AdminUser] = resp.PachToken
-	config.WritePachTokenToConfig(resp.PachToken)
+	config.WritePachTokenToConfig(RootToken)
 
 	// Wait for the Pachyderm Auth system to activate
 	require.NoError(tb, backoff.Retry(func() error {
-		if isAuthActive(tb, true) {
+		if isActive, err := client.IsAuthActive(); err != nil {
```

Comment:
I actually didn't know we had this :joy:. I went and looked it up, and it looks like Yusuf added it for the S3 gateway.

I'm pretty sure I wrote at least some code that checks whether auth is active by calling `WhoAmI`. It's out of scope for the PR, but FWIW there are probably calls to `WhoAmI` that discard the result and should maybe be calling this instead.